src/Castle.Windsor.Extensions.DependencyInjection/SubSystems/DependencyInjectionNamingSubsystem.cs

```diff
@@ -123,5 +123,11 @@ public override IHandler GetHandler(Type service)
 			return null;
 		}
 
+		protected override void InvalidateCache()
+		{
+			handlerListsRegistrationOrderByTypeCache.Clear();
+			handlerListsPriorityOrderByTypeCache.Clear();
+			base.InvalidateCache();
+		}
```

Comment:
Can you add a unit test to show what this is fixing and so we don't get that defect back.

Reply:
Yes, but not until next week.
Good point, Thabks